### PR TITLE
[1822Africa] Allow placing Gold Mine if corporation is out of tokens

### LINKED ISSUE #9676: 1822Africa - Cannot use P14 Power
https://18xx.games/game/134428?action=458 

Possibly triggered because the company is out of tokens to place. It worked correctly here: https://18xx.games/game/135568?action=463

### PR DESCRIPTION
**Pull request by vandamm**
Fixes #9676

### Before clicking "Create"

- [x] Branch is derived from the latest `master`
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

### CODE REVIEW

File: lib/engine/game/g_1822_africa/step/token.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  module Game
+    module G1822Africa
+      module Step
+        class Token < G1822::Step::Token
+          def actions(entity)
+            actions = super.dup
+
+            actions << 'choose_ability' if need_choose_ability?(entity, actions) && !actions.include?('choose_ability')
+
+            actions << 'pass' if actions.any? && !actions.include?('pass')
```
Comment (michaeljb) [inline lib/engine/game/g_1822_africa/step/token.rb:15]: The JS produced by Opal for `any?` without a block is much less efficient than `!empty?`. (Unfortunately there are also a bunch of other places in the code doing this that need to be cleaned up too)

Comment (vandamm) [reply lib/engine/game/g_1822_africa/step/token.rb:15]: Can we make rubicon check for that?

Comment (michaeljb) [reply lib/engine/game/g_1822_africa/step/token.rb:15]: Not with any cops/configurations I’ve been able to find

Comment (vandamm) [reply lib/engine/game/g_1822_africa/step/token.rb]: @michaeljb changed it

**Pull request merged**